core/include/detray/masks/ring2D.hpp

```diff
@@ -128,6 +128,21 @@ class ring2D {
         return {-r_bound, -r_bound, -env, r_bound, r_bound, env};
     }
 
+    /// @brief Calculates the coordinates of the vertices.
+    ///
+    /// @param bounds the boundary values for this shape.
+    ///
+    /// @returns a container of vertices. If the shape contains
+    /// no vertices an empty container will be returned.
+    template <typename point3_container_t,
+              template <typename, std::size_t> class bounds_t,
+              typename scalar_t, std::size_t kDIM,
+              typename std::enable_if_t<kDIM == e_size, bool> = true>
+    DETRAY_HOST inline point3_container_t local_vertices(
+        const bounds_t<scalar_t, kDIM> &) const {
+        return {};
```

Comment:
How will ring and cylinder shape be handled eventually?

Reply:
So are we going to leave this function empty? I am fine but it doesn't look neat

Reply:
We could add it in a later PR for completion, but right now I don't think its needed. A possibility would be to just throw an exception if this functions is ever called

Reply:
I made some changes so the `local_vertices` function has been removed from the masks. Instead, the visualization code uses functions in tools/generator.hpp. Furthermore, it might be possible to avoid the special cases regarding rotation for discs by using the annulus type in actsvg (which draws the shape as a hollow polygon), if this is needed.